src/server.js

```diff
@@ -824,6 +832,11 @@ function performAuthentication(data) {
 		return;
 	}
 
+	// Prevent empty headers from creating users with blank / no username
+	if (data.user === "") {
```

Comment:
As far as I can tell this only applies in the case where headerauth is active. Anything else simply fails the auth as it should, right?

So then put it where it belongs. A few lines above when you actually get the header value, if it's empty respond with a failure in the log and return. That's a misconfiguration most likely so it would be nice to give the admin an indication what's wrong.

Reply:
Better yet, ignore it here, fail the auth inside of header auth as that's an invalid input value to that.
That means every backend works the same